### ISSUE TITLE
Webpack build error in algosdk 1.8.0

### ISSUE THREAD

**Issue by DJRHails** [algorand/js-algorand-sdk#244]
Using yarn berry to build a React application with an algosdk dependency causes the following error.

```sh
Δ yarn set version from sources
Δ yarn 
Δ yarn build
.yarn/cache/algosdk-npm-1.8.0-0f9313ab3a-3db2b154e4.zip/node_modules/algosdk/src/logicsig.js
Module not found: Recursion in resolving
```

Currently the fix is the prevent the version 1.8.0 from being installed.
```json
    "algosdk": ">=1.6.2 <1.8.0",
```

**Comment by janmarcano** [comment]
Hopping in to confirm that I've faced the same issue with npm, I had to roll back to 1.7.2 in the meantime as well.
```
ERROR in ./node_modules/algosdk/src/logicsig.js
Module not found: Error: Recursion in resolving
...
ERROR in ./node_modules/algosdk/src/client/v2/algod/sendRawTransaction.js
Module not found: Error: Recursion in resolving
...
ERROR in ./node_modules/algosdk/src/logicTemplates/templates.js
Module not found: Error: Recursion in resolving
...
```

**Comment by jasonpaulos** [comment]
Thanks for reporting this, I'm looking into it now. Does the error happen when installing or executing code?

**Comment by janmarcano** [comment]
On my side, it happened during webpack bundling.

**Comment by jasonpaulos** [comment]
@janmarcano thanks for the info. It looks like due to some code changes and updates to our dependencies, our module requires some additional webpack configuration to work in 1.8.0. Could you let me know if the following works for you?

Install additional dependencies: `npm i --save assert readable-stream buffer`

Add to webpack.config.js:
```javascript
module.exports = {
    resolve: {
        alias: {
            assert: require.resolve("assert/"),
            stream: require.resolve("readable-stream"),
            buffer: require.resolve("buffer"),
        }
    }
};
```

**Comment by DJRHails** [comment]
Can also confirm it was during bundling / build for production.
I've been using CRA's react-script https://github.com/facebook/create-react-app/tree/master/packages/react-scripts so modifying `webpack.config.js` would require ejecting which isn't the end of the world but a bit of an added frustration.

**Comment by jasonpaulos** [comment]
@DJRHails I believe you can achieve the same thing without ejecting from the react-scripts by creating/editing the `metro.config.js` file to have this:

```javascript
module.exports = {
  resolver: {
    extraNodeModules: {
      assert: require.resolve("assert/"),
      stream: require.resolve("readable-stream"),
      buffer: require.resolve("buffer"),
    },
  },
};
```

Let me know if that works for you.

**Comment by janmarcano** [comment]
@jasonpaulos installing the dependencies and adding the aliases seems to work.
I tested my whole application briefly and there were no issues.

**Comment by jasonpaulos** [comment]
@janmarcano great, thanks for letting me know. That confirms what I thought the problem was. For some background, `assert` is required because our code uses it, however `readable-stream` is needed because our dependency [`keccak`](https://www.npmjs.com/package/keccak) uses streams, and `buffer` is needed for `readable-stream` to work.

So the problem isn't just with our library. I'll think about how we can address this issue.

**Comment by DanBurton** [comment]
Bump. I was about to report the same issue, but I see one already exists. Here's my process for reproducing it:

```bash
npx create-react-app algosdk-react
cd algosdk-react
npm install --save algosdk
# + algosdk@1.8.0

# add to top of src/App.js
import algosdk from 'algosdk';

# add to top of function App in src/App.js, just to make sure algosdk actually gets compiled in
console.log(algosdk.algosToMicroalgos(10)); 

# then run this
npm run start
```

I see the same errors as others have mentioned. `Cannot find module 'assert'`. (Even after doing `npm install --save assert` as well.)

I tried a modified version of the suggestion above using [craco](https://github.com/gsoft-inc/craco/blob/master/packages/craco/README.md), which sort of works.

```bash
npm i --save assert readable-stream buffer
npm install --save @craco/craco
# update package.json to 
# create craco.config.js as below:
```

```js
// craco.config.js
const path = require('path');

module.exports = {
  webpack: {
    alias: {
      assert: path.resolve(__dirname, './node_modules/assert'),
      stream: path.resolve(__dirname, './node_modules/readable-stream'),
      buffer: path.resolve(__dirname, './node_modules/buffer'),
    }
  }
}
```

It is able to get past the module resolution issues, but compilation still fails with `TypeError: Cannot read property 'isPromise' of undefined`, which I assume is because the way I am "aliasing" node modules is wrong.

**Comment by jasonpaulos** [comment]
I just merged a PR which should fix these webpack issues without requiring modifying any configs. It will be published in version 1.8.1.

In the meantime, here's a preview of it if anyone would like to verify it works: 
[algosdk-1.8.1-beta.1.tgz.zip](https://github.com/algorand/js-algorand-sdk/files/5569132/algosdk-1.8.1-beta.1.tgz.zip). You will have to extract the algosdk-1.8.1-beta.1.tgz file from the zip (github doesn't support uploading that file type) and then install with with `npm i algosdk-1.8.1-beta.1.tgz`.
